polus-feature-heatmap-pyramid-plugin/src/main.py

```diff
@@ -0,0 +1,348 @@
+from bfio import BioWriter
+import bioformats
+import javabridge as jutil
+import argparse, logging, time, imagesize, os, re
+from filepattern import FilePattern
+import numpy as np
+from pathlib import Path
+
+STITCH_VARS = ['file','correlation','posX','posY','gridX','gridY'] # image stitching values
+STITCH_LINE = "file: {}; corr: {}; position: ({}, {}); grid: ({}, {});\n"
+
+# Initialize the logger
+logging.basicConfig(format='%(asctime)s - %(name)-8s - %(levelname)-8s - %(message)s',
+                    datefmt='%d-%b-%y %H:%M:%S')
+logger = logging.getLogger("main")
+logger.setLevel(logging.INFO)
+
+def mean(data_list):
+    """ Fast mean
+    
+    Calculates the artithmetic mean. This is faster than the method of the same name
+    in the statistics package (1.5-10x speed improvement based on list size).
+
+    Inputs:
+        data_list - a list of floats
+    Outputs:
+        val - The arithmetic mean of data_list
+    """
+    val = sum(data_list)/len(data_list)
+    return val
+
```

Comment:
So generally this is all well organized and I like how you create a definition for essentially an error check. We talked about implementing more than mean (should be trivial to do) once we had more time. Perhaps get somebody to add, SD, (SE if rows < 20), skewness, kurtosis, entropy, median, mode, and quartiles?